docs/includes/fundamentals/write-operations/WriteOperationsTest.php

```diff
@@ -329,4 +330,188 @@ public function testModelPositional(): void
+
+        $data = [
+            [
+                'performer' => 'Mitsuko Uchida',
+                'venue' => 'Carnegie Hall',
+                'genres' => ['classical'],
+                'ticketsSold' => 2121,
+                'performanceDate' => new UTCDateTime(Carbon::create(2024, 4, 1, 20, 0, 0, 'EST')),
+            ],
+            [
+                'performer' => 'Brad Mehldau',
+                'venue' => 'Philharmonie de Paris',
+                'genres' => [ 'jazz', 'post-bop' ],
+                'ticketsSold' => 5745,
+                'performanceDate' => new UTCDateTime(Carbon::create(2025, 2, 12, 20, 0, 0, 'CET')),
+            ],
+            [
+                'performer' => 'Billy Joel',
+                'venue' => 'Madison Square Garden',
+                'genres' => [ 'rock', 'soft rock', 'pop rock' ],
+                'ticketsSold' => 12852,
+                'performanceDate' => new UTCDateTime(Carbon::create(2025, 2, 12, 20, 0, 0, 'CET')),
+            ],
+            [
+                'performer' => 'The Rolling Stones',
+                'venue' => 'Soldier Field',
+                'genres' => [ 'rock', 'pop', 'blues' ],
+                'ticketsSold' => 59527,
+                'performanceDate' => new UTCDateTime(Carbon::create(2024, 6, 30, 20, 0, 0, 'CDT')),
+            ],
+        ];
+        Concert::insert($data);
+
+        $this->assertEquals(4, Concert::count());
+
+        $id = Concert::first()->id;
+
+        // begin model delete by id
+        Concert::destroy($id);
+        // end model delete by id
+
+        $this->assertEquals(3, Concert::count());
+    }
+
+    public function testModelDeleteModel(): void
+    {
+        require_once __DIR__ . '/Concert.php';
+        Concert::truncate();
+
+        $data = [
+            [
+                'performer' => 'Mitsuko Uchida',
+                'venue' => 'Carnegie Hall',
+            ],
+        ];
+        Concert::insert($data);
+
+        // begin delete one model
+        $concert = Concert::first();
+        $concert->delete();
+        // end delete one model
+
+        $this->assertEquals(0, Concert::count());
+    }
+
+    public function testModelDeleteFirst(): void
+    {
+        require_once __DIR__ . '/Concert.php';
+        Concert::truncate();
+
+        $data = [
+            [
+                'performer' => 'Mitsuko Uchida',
+                'venue' => 'Carnegie Hall',
+            ],
+            [
+                'performer' => 'Brad Mehldau',
+                'venue' => 'Philharmonie de Paris',
+            ],
+            [
+                'performer' => 'Billy Joel',
+                'venue' => 'Madison Square Garden',
+            ],
+            [
+                'performer' => 'The Rolling Stones',
+                'venue' => 'Soldier Field',
+            ],
+        ];
+        Concert::insert($data);
+
+        // begin model delete one fluent
+        Concert::where('venue', 'Carnegie Hall')
+            ->orderBy('_id')
```

Comment:
I don't get the point of ordering for deleting. In fact, it's ignored:
https://github.com/mongodb/laravel-mongodb/blob/95c5b4fac69c2bd5ea01367e62d14dd5278e7513/src/Query/Builder.php#L787-L814

Reply:
I see, so if the where() query filter matched multiple documents, wouldn't it be important to specify the sort order such that it's known which one will surface when calling limit(1), since that's the one that will be deleted? Does that mean that you can't use indexes with sort orders for the query filter part of the delete operation?

If not possible, I should probably include that information.

Reply:
That's currently not possible. That could be a new feature PHPORM-169

Reply:
> Does that mean that you can't use indexes with sort orders for the query filter part of the delete operation?

Note: from the server's perspective, there is no way to specify an order with a [delete](https://www.mongodb.com/docs/manual/reference/command/delete/) operation.

Indexes can certainly be used (and [hinted](https://www.mongodb.com/docs/manual/reference/command/delete/#specify-hint-for-delete-operations)). In the case of a [sparse index](https://www.mongodb.com/docs/manual/core/index-sparse/) that might actually have an impact on which document(s) get selected for the deletion.